Core/TermsOfService.swift

```diff
@@ -20,125 +20,84 @@
-            .c: 0,
-            .d: 1,
-            .e: 2
-            ]
-
-        static let derivedScoreAsPractices: [Int: PrivacyPractices] = [
-            -1: .good,
-            0: .mixed,
-            1: .poor
-            ]
-
-    }
-
-    public let classification: Classification?
-    public let score: Int
-    public let goodReasons: [String]
-    public let badReasons: [String]
-
-    public var hasReasons: Bool {
-        return !goodReasons.isEmpty || !badReasons.isEmpty
-    }
-
-    public var hasUnknownPractices: Bool {
-        return !hasReasons
+        case reasons = "match"
+        case classification = "class"
     }
-
-    public var derivedScore: Int {
+    
+    // see https://github.com/duckduckgo/duckduckgo-privacy-extension/blob/e42533/shared/js/background/privacy-practices.es6.js#L65
+    var summary: PrivacyPractices.Summary {
+        
         if let classification = classification {
-            return Lookups.classificationScores[classification]!
+            return TermsOfService.classificationSummaries[classification]!
         }
-
-        return normalizeScore()
-    }
-
-    public func privacyPractices() -> PrivacyPractices {
-        guard !hasUnknownPractices else { return .unknown }
-
-        var practices: PrivacyPractices?
-        if let classification = classification {
-            practices = Lookups.classificationAsPractices[classification]
-        } else if !goodReasons.isEmpty && !badReasons.isEmpty {
+        
+        if hasGoodReasons && hasBadReasons {
             return .mixed
-        } else {
-            practices = Lookups.derivedScoreAsPractices[normalizeScore()]!
         }
-
-        if let practices = practices {
-            return practices
+        
+        if score < 0 {
+            return .good
+        } else if score == 0 && (hasGoodReasons || hasBadReasons) {
+            return .mixed
+        } else if score > 0 {
+            return .poor
         }
-
+        
         return .unknown
     }
-
-    private func normalizeScore() -> Int {
-        // extensions JS uses Math.sign(score)
-        if score < 0 { return -1 }
-        if score > 0 { return 1 }
-        return 0
-    }
-
-    public enum Classification: String, Decodable {
-        case a, b, c, d, e
-
-        public init(from decoder: Decoder) throws {
-            let container = try decoder.singleValueContainer()
-            let lowercaseString = try container.decode(String.self).lowercased()
-            guard let classification = Classification(rawValue: lowercaseString) else {
-                let context = DecodingError.Context(codingPath: [], debugDescription: "Classification string didn't match")
-                throw DecodingError.typeMismatch(Classification.self, context)
-            }
-            self = classification
+    
+    // see https://github.com/duckduckgo/duckduckgo-privacy-extension/blob/e42533/shared/js/background/privacy-practices.es6.js#L20
+    var derivedScore: Int {
```

Comment:
@andrey-p this is where the new score for the terms of service is calculated - should look pretty similar